### PR TITLE
Update build and release systems

### PR DESCRIPTION
**Pull request by fiorix**
This diff removes the build_tarballs.sh and the rpm/Makefile files and
moves their logic into the main Makefile.

The build system places binary files in the build directory. The release
system copy those files over to the release directory, preparing it for
travis to pick up.

Examples:

make binary_deb binary_rpm VERSION=1.0
make clean binary_tar VERSION=1.0 GOOS=linux GOARCH=amd64
make distclean release VERSION=1.0

Deleted the debian/copyright file because it was outdated; I think this
could be automated with debmake but didn't spend time on it.

There's other minor changes like removing trailing space from files,
updating and adding documentation.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,38 +1,154 @@
...
+
+# binary_tar creates tarball of binary distribution.
+binary_tar: binary_dist
+	[ -d build/tgz ] || mkdir build/tgz
+	cd build && $(TAR) czf tgz/$(NAME)_$(GOOS)_$(GOARCH)-$(VERSION).tar.gz $(DIST_NAME)
+	rm -rf $(DIST_DIR)
+
+# binary_zip creates zip of binary distribution.
+binary_zip: binary_dist
+	[ -d build/zip ] || mkdir build/zip
```
Comment (skogtwin) [inline Makefile]: this test (as well as others blow) is unnecessary: mkdir of exisiting directory is a no-op.

File: Makefile [hunk 2]
```diff
@@ -1,38 +1,154 @@
...
+# binary_rpm creates rpm package.
+#
+# Requires GOPATH and dependencies available to compile nvdtools.
+# Must set version to build: make binary_rpm VERSION=1.0
+binary_rpm: archive_tar
+	[ -d build/rpm/SOURCES ] || mkdir -p build/rpm/SOURCES
+	mv build/tgz/$(DIST_NAME).tar.gz build/rpm/SOURCES/
+	rpmbuild -ba \
+		--define="_topdir $(PWD)/build/rpm" \
+		--define="_version $(VERSION)" \
```
Comment (skogtwin) [inline Makefile]: We probably need to start maintaining a changelog. The we could grep the version from it... Just a thought, not related to this commit specifically.

**Comment by fiorix** [comment]
> Why did you delete debian/copyright? Don't we need it for legal reasons or what not?

The file is auto-generated by something (debmake?) and contain a list of source code files and their licenses. I didn't want to update it manually, couldn't find an easy way to re-generate. If there is one we should do it. I don't think it impacts licensing much, given we are not shipping the LICENSE file as part of the docs.

File: Makefile (hunk 1 above)
Comment (fiorix) [reply Makefile]: It actually fails:

```
vagrant@ubuntu-bionic:/vagrant/nvdtools$ mkdir foo
vagrant@ubuntu-bionic:/vagrant/nvdtools$ mkdir foo
mkdir: cannot create directory ‘foo’: File exists
vagrant@ubuntu-bionic:/vagrant/nvdtools$ echo $?
1
```

Comment (skogtwin) [reply Makefile]: mkdir -p, sorry if it wasn't clear from my previous comment

Comment (skogtwin) [reply Makefile]:
```
-p, --parents
              no error if existing, make parent directories as needed
```
(from man page)

File: Makefile (hunk 2 above)
Comment (fiorix) [reply Makefile]: Both rpm and debian packages need changelog. We should probably auto-generate them from the commit log or something along those lines.

The version needs more love: we can now easily build packages with any version by passing VERSION to make. This is how we use git tags in .travis.yml for automated releases.

What is missing is having a -version in all binaries.

**Comment by fiorix** [comment]
@skogtwin @n1chre any idea how we can test the travis release before merging this?

**Comment by skogtwin** [comment]
> @skogtwin @n1chre any idea how we can test the travis release before merging this?

I don't think we can... Try tagging the diff in PR with semver, if it doesn't work, then, well, testinprod (TM)

**Comment by fiorix** [comment]
Removed all -d checks in favor of mkdir -p

**Pull request merged**